index.js

```diff
@@ -294,10 +294,10 @@ function createHtmlFileList(files, dir, useIcons, view) {
 
     path.push(encodeURIComponent(file.name));
 
-    var date = file.stat && file.name !== '..'
+    var date = file.stat && file.stat.mtime && file.name !== '..'
       ? file.stat.mtime.toLocaleDateString() + ' ' + file.stat.mtime.toLocaleTimeString()
       : '';
-    var size = file.stat && !isDir
+    var size = file.stat && file.stat.size && !isDir
```

Comment:
Is there a bug on Node.js regarding this? The `fs.stat` cannot return something other than a stat object unless it provides an error. Or is there missing logic in here regarding the error from `fs.stat` not getting propogated? We should probably fix the reason the stat is missing rather than the symptom.